src/NexusMods.DataModel/Diagnostics/DiagnosticManager.cs

```diff
@@ -78,43 +80,41 @@ private async Task<Diagnostic[]> GetLoadoutDiagnostics(Loadout.ReadOnly loadout,
 
         try
         {
-            var nested = await diagnosticEmitters
-                .OfType<ILoadoutDiagnosticEmitter>()
-                .SelectAsync(async emitter =>
-                {
-                    var start = DateTimeOffset.UtcNow;
-
-                    try
-                    {
-                        var res = await emitter
-                            .Diagnose(loadout, cancellationToken)
-                            .ToListAsync(cancellationToken);
+            List<Diagnostic> diagnostics = [];
+            
+            await Parallel.ForEachAsync(diagnosticEmitters.OfType<ILoadoutDiagnosticEmitter>(), cancellationToken, async (emitter, token) =>
```

Comment:
The caller of `GetLoadoutDiagnostics` (line 60) is using `SelectMany` which doesn't guarantee order, and isn't passing in cancellation tokens. 

We should switch it to `R3.Observable` with `ToObservable()` and then use `SelectAwait` with `AwaitOperation.Switch`, cancellation should hopefully work better here, but we can also use `AwaitOperation.ThrottleFirstLast` if that is not the case.

Reply:
I had a look through this code as well.
This seems to check out.

To be specific, `SelectMany` on an observable returns items in the order that they complete in. If you're running on multiple threads, and a context switch happens (or `await GetLoadoutDiagnostics` stalls for longer for whatever reason), it is possible for the items to be emitted out of order; meaning we may receive diagnostics for an earlier loadout state.

In practice the chance of that happening though, is actually basically 0, especially given we've a Throttle right before that. However, it's better to be technically correct rather than correct simply due to external circumstance.